consumers/campaigner/main.go

```diff
@@ -66,128 +114,149 @@ func (h *MessageHandler) HandleMessage(m *nsq.Message) error {
 			logrus.WithFields(logrus.Fields{
 				"user_id":     msg.UserID,
 				"segment_ids": msg.SegmentIDs,
-			}).WithError(err).Error("Unable to fetch subscribers.")
-			break
+			}).WithError(err).Error("unable to fetch subscribers.")
+			return nil
 		}
 
 		if len(subs) == 0 {
```

Comment:
Shouldn't you check param next in pagination to break the loop

Reply:
@gudgl GetSubs method is not pagination